src/utils/capitalize/index.js

```diff
@@ -0,0 +1,2 @@
+export { default } from './capitalize'
+export * from './capitalize'
```

Comment:
Despite how utils directory looks in the end, I find this exporting part very repetitive and an extra thing to maintain.

Maybe, can we start from what you have found unclear or confusing about the previous folder structure?
I always tend to perceive folder-grouped files like a functionality related to a single module. So it makes sense when a module `A` requires `funcA`, `funcB` and `funcC` to work. Then grouping it in a single directory is a good thing to do.

At the moment, most of the utils are self-contained, and for those who indeed require some unique helpers there's a directory already ("math"). For the small library which it is this change appears an overkill.

Reply:
This is indeed very repetitive, you could achieve the same effect without this repetitive syntax by just calling your actual main file in the folder just index.js but the problem with that is that every tab in your editor says index.js and you have no idea what file you're in just by looking at the tab.

The reason why this grouping is useful is because usually you have you main module but then you have a lot of things that go with it. Examples: test-file, type-definition-files (if you ever wanna add ts support you can just put an index.d.ts file in every folder with your exposed api), storybook (for react modules), just the kind of stuff that is self-contained to that module specifically. (to be honest I've just seen some projects do that, especially with react components, and I found it pretty nifty)

It just makes for less of a file-wall to look at from the root utils directory, at the moment it basically is half the amount of things to look at when you open the utils directory and you don't really lose any information because you're still seeing the module names the only thing you're losing is knowledge about the complementary files (in your case just tests for now).

Reply:
It would be great if there was a codemod which can resolve the module import by attempting to import the `[dirName].js`. This way we could omit this verbose exports for each module.

Reply:
take a look at this [mui-components](https://github.com/mui-org/material-ui/tree/master/packages/material-ui/src)

Reply:
This is really pro-human-readability in the editor, completely unrelated to any runtime effect. I know there's this [nice-index](https://github.com/joshwcomeau/nice-index). Maybe there is a webpack plugin that does that?

Reply:
Let's give it a shot to find such webpack or babel plugin then.

Reply:
https://www.npmjs.com/package/directory-named-webpack-plugin

Reply:
@MatthiasMargot here it is https://www.npmjs.com/package/directory-named-webpack-plugin
Do you mind setting it up? Thanks!

Reply:
sure, will do :))